src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts

```diff
@@ -79,6 +82,8 @@ export class CloneRepoButtonComponent implements OnInit, OnChanges {
 
         this.useSsh = this.localStorage.retrieve('useSsh') || false;
         this.localStorage.observe('useSsh').subscribe((useSsh) => (this.useSsh = useSsh || false));
+
+        this.getVSCTokenIfNotPresent();
```

Comment:
Does this work here in call cases? Or should we move it into the subscribe block above?

Reply:
It even needs both the user and the profile info. So you probably need to refactor it a bit to be able to `zip` the two together to subscribe to a `(user, profileInfo)` tuple and then set all the necessary attributes.

Reply:
On it, thanks for the feedback!